rfd/0097-teleport-connect-usage-metrics.md

```diff
@@ -0,0 +1,179 @@
+The only issue with anonymizing events client-side is lack of `cluster id` that is kept in `Auth Server`. To remedy
+this, when the app starts and retrieves cluster information, it should also retrieve the `cluster id`, create an
+anonymizer and store it in the cluster struct.
+
+### Storing events
+
+Batches of anonymized events will be sent to a public endpoint in `prehog` (intended for use only by Connect) that
+translates them into the PostHog's data model.
+
+Connect events will share the same project with clusters and website events. It will allow to perform queries that need
+both sources of data, like calculating what is the percentage of users logging to a cluster with Connect.
+
+Some event properties, like OS can be saved as a user property. These properties are
+then [stored directly on each event](https://posthog.com/docs/how-posthog-works/data-model#person). For example, when
+the first emitted event sets a user property `os: windows`, each next event will have this property set.
+
+To differentiate events coming from multiple application instances, each event needs to have `distinct_id` field. It
+will be supplied with UUID generated by Connect with `connect.` prefix. The value will be created on the start and
+stored in a file in the _App Data_directory, so it will not change between restarts.
+
+> **NOTE:** As wrote above, Connect events are tied to the application instance (or just client machine). It means that
+> PostHog's _Person_ for Connect and for cluster will be a different thing. They should not be merged.
+
+### User agreement
+
+On the start, Teleport Connect will ask user to opt in to volunteer anonymized metrics and usage-data with standard
+message "Are you OK sending anonymized usage data about Teleport Connect? This will help us to improve product".
+
+If the user refuses, Connect will not send any usage data.
+
+### How will collecting metrics support product development?
+
+In the initial version, it should help with getting answers to the following questions:
+
+#### How many unique users download and use Teleport Connect today?
+
+To answer the first part of the question, download counts from `goteleport.com/download` are needed. These will be
+collected basing on access logs from CloudFront CDN.
+
+To calculate how many users use Teleport Connect on a daily basis, a metric like DAU (Daily Active Users) can be used.
+This metric can be based on a specific event, but in this case it should be calculated using any event. For example,
+user logged only once in a given day to refresh certs for a DB proxy connection - such user can be considered as active.
+
+#### What features are the most popular?
+
+Usage of each feature will be measured basing on events from the [events](#events) section. They will allow to generate
+various statistics, like the most common kinds of connections or just show the usage of particular features like Access
+Requests.
+
+#### Where users struggle the most?
+
+//TODO
```

Comment:
I would remove this section if we're still not sure. 

In theory, we could capture a few user journeys, translate them to events, and see where in the journey they drop off, if ever. 
Ex: If there are 7 event steps in a journey, and 80% of users stop at event 4, then event 5 is the one to look at.

Reply:
Yeah, I'm going to remove it. 

Capturing user journeys sounds good, but we would need to add more events (so this is out of scope for now).